submissions/lhousaine/index.js

```diff
@@ -1,102 +1,178 @@
 const fs = require('fs');
 const readline = require('readline');
+const os = require('os');
+const {
+  Worker,
+  isMainThread,
+  parentPort,
+  workerData,
+} = require('worker_threads');
+
+if (isMainThread) {
+  const inputFile = '../../input.txt';
+  const outputFile = 'output.txt';
+  const numCPUs = os.cpus().length;
+  const { size } = fs.statSync(inputFile);
+  const chunkSize = Math.max(Math.floor(size / numCPUs), 1);
+
+  const fd = fs.openSync(inputFile);
```

Comment:
Won't this load the whole file in memory ?

Reply:
@aboullaite no, it returns only a file descriptor with read mode, to be used bellow in the while loop to calculate chunks boundaries indexes that will be processed by each worker.